Remotion/ObjectBinding/Web.ClientScript/Scripts/BocList.ts

```diff
@@ -492,116 +498,125 @@ class BocList
-    var realTableWidth = realTable[0].offsetWidth;
-    var previousRealTableWidth = realTable.data("bocListPreviousRealTableWidth");
+    var realTable = scrollableContainer.querySelector<HTMLTableElement>(':scope>table')!;
+    var realTableWidth = BocList.getComputedWidth(realTable);
+    var previousRealTableWidth = parseInt(realTable.dataset.bocListPreviousRealTableWidth!)
     if (previousRealTableWidth == realTableWidth)
       return;
-    realTable.data("bocListPreviousRealTableWidth", realTableWidth);
+    realTable.dataset.bocListPreviousRealTableWidth = realTableWidth.toString()
   
-    var realTableHead = realTable.eq(0).find('thead');
-    var realTableHeadRow = realTableHead.children().eq(0);
-    var realTableHeadRowChildren = realTableHeadRow.children();
+    var realTableHead = realTable.querySelector('thead')!;
+    var realTableHeadRow = realTableHead.children[0];
+    var realTableHeadRowChildren = realTableHeadRow.children;
   
-    var fakeTableHeadContainer = scrollableContainer.parent().children('div.bocListFakeTableHead').first();
-    var fakeTableHead = fakeTableHeadContainer.find('thead');
-    var fakeTableHeadRow = fakeTableHead.children().eq(0);
-    var fakeTableHeadRowChildren = fakeTableHeadRow.children();
+    var fakeTableHeadContainer = scrollableContainer.parentNode!.querySelector<HTMLTableElement>(':scope>div.bocListFakeTableHead')!;
+    var fakeTableHead = fakeTableHeadContainer.querySelector('thead')!;
+    var fakeTableHeadRow = fakeTableHead.children[0];
+    var fakeTableHeadRowChildren = fakeTableHeadRow.children;
   
     // store cell widths in array
     var realTableHeadCellWidths = new Array();
-    realTableHeadRowChildren.each(function (this: HTMLElement, index)
+    Array.from(realTableHeadRowChildren).forEach(function (element: Element, index: number)
     {
-      realTableHeadCellWidths[index] = $(this).width();
+      realTableHeadCellWidths[index] = BocList.getComputedWidth(element as HTMLElement);
     });
   
     // apply widths to fake header
-    fakeTableHeadRowChildren.width(function (index, itemWidth)
+    Array.from(fakeTableHeadRowChildren).forEach((element: Element, index: number) =>
     {
-      var width = realTableHeadCellWidths[index];
-      return width;
+      (element as HTMLElement).style.width = realTableHeadCellWidths[index] + 'px';
     });
-  
-    var fakeTableHeadWidthContainer = fakeTableHeadContainer.children ('div').first();
-    fakeTableHeadWidthContainer.width(realTableWidth);
-    var fakeTableHeadContainerHeight = fakeTableHeadContainer.height();
-    scrollableContainer.css({ top: fakeTableHeadContainerHeight});
-    realTable.css({ 'margin-top': fakeTableHeadContainerHeight * -1 });
-  
-    fakeTableHeadContainer.show();
+
+    var fakeTableHeadWidthContainer = fakeTableHeadContainer.querySelector<HTMLElement> (':scope>div')!;
+    fakeTableHeadWidthContainer.style.width = realTableWidth + 'px'
+    fakeTableHeadContainer.style.display = 'block';
```

Comment:
Why was this moved up?

Reply:
Excellent question; `getComputedStyle` always returns 0 for height and width if the element's display style is `none` or `inline`, so it needs to become a `block` before we get its height and width